### PR TITLE
TR-550: Edit a Snippet

### PR DESCRIPTION
**Pull request by bkemper**
**Prerequisites**

To test, you will need to pull the latest SparkPost/hamurai for the /api/labs/snippets endpoints.

```sh
$ cd hamurai && git pull & hamurai --tags openresty
```

### CODE REVIEW

File: src/pages/snippets/EditPage.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      return <RedirectAndAlert to="/snippets" alert={{ type: 'error', message: error.message }} />;
+    }
+
+    if (loading) {
+      return <Loading />;
+    }
+
+    return (
+      <Page
+        title="Edit Snippet"
```
Comment (ewandennis) [inline src/pages/snippets/EditPage.js]: Should we say `View Snippet` when `!canModify`?

Comment (ewandennis) [reply src/pages/snippets/EditPage.js]: Actually in the template editor, we use template ID in the page title. Maybe more useful than "Edit Snippet"? *shrug*

Comment (bkemper) [reply src/pages/snippets/EditPage.js]: Name is not a required field in the API, so if we are okay with the id being the alternative, then I'll change that.

File: src/pages/snippets/components/IdentifierHelpText.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import ExternalLink from 'src/components/externalLink/ExternalLink';
+import { LINKS } from 'src/constants';
+
+const IdentifierHelpText = () => (
+  <React.Fragment>
+    Use this unique identifier to {(
+      <ExternalLink to={LINKS.SNIPPET_SUBSTITUTION_REFERENCE}>
```
[the remark this answers is not in the document]
Comment (bkemper) [reply src/pages/snippets/components/IdentifierHelpText.js]: Yep, I should have provided a note that the link will not work until that PR is merged.

File: src/pages/snippets/EditPage.js (hunk 1 above)
Comment (bkemper) [reply src/pages/snippets/EditPage.js]: Addressed... 80c2b6d

**Comment by bkemper** [comment]
@ewandennis I addressed all feedback and this is ready for final review.

**Pull request closed without merging**